### DISCUSSION TITLE
With GTK4 support transmission should perhaps implement proper GNOME theming support

### DISCUSSION THREAD

**Discussion by a-lameira** [transmission/transmission#5239] [Ideas]
With the implementation of GTK4 support and the start of distribution of packages in large distros built against GTK4, an issue has arisen: the GTK4 version of Transmission does not support any themes at least on the GNOME platform.

Unlike GTK3 apps, which could easily be themed with custom themes like adw-gtk3, GTK4 apps require proper theme support to integrate correctly with the current version of the main GTK4-based DE, GNOME, which is default for major Linux distros like Ubuntu, Fedora and Debian.

I think that a discussion about updating the program interface to the new reality of GTK in Desktop Environments would be more than appropriate. While I find the current Transmission interface adequate, the fact is that the design reflects the days of GTK2.

There are already some approaches to this discussion, such as #3038 , #3719 , #5140 , #5224 and #5137 , but I think we can go a step further and properly open up the discussion about themes and GTK4. Unfortunately, I am not an expert on GTK4 or themes, but I have some basic contributions to the discussion.

Since version 42 of GNOME, there is official support for system-wide style management, [as described here](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/class.StyleManager.html), through the relevant API. Support for Light and Dark modes are part of the GNOME guidelines, as posted [here](https://developer.gnome.org/hig/guidelines/ui-styling.html). As much has been said, the current official theme for GTK4 applications in the GNOME platform is provided by Libadwaita. Probably using Libadwaita would involve a greater or lesser degree of UI redesign. Another alternative would be to ship theme options with the application, as many programs do, leaving the user the choice through Transmission's interface.

**Comment by ile6695** [comment]
> ...GTK4-based DE, GNOME, which is default for major Linux distros like Ubuntu, Fedora and Debian.

~~Currently there are no Debian releases that contain [GTK4](https://tracker.debian.org/pkg/gtk4), so it cannot be the default. Unless you meant that just GTK3/4 is the default, which is true.~~
With the release of Bookworm last month, GTK4 is now included there as well :tada:
